src/main/java/org/gridsuite/loadflow/server/utils/SpecificationUtils.java

```diff
@@ -0,0 +1,134 @@
+            String stringValue = (String) value;
+            String escapedFilterValue = EscapeCharacter.DEFAULT.escape(stringValue);
+            if (escapedFilterValue == null) {
+                throw new UnsupportedOperationException("Filter text values can not be null");
+            }
+            // this makes equals query work with enum values
+            Expression<String> stringExpression = expression.as(String.class);
+            return switch (filter.type()) {
+                case CONTAINS -> criteriaBuilder.like(criteriaBuilder.upper(stringExpression), "%" + escapedFilterValue.toUpperCase() + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
+                case STARTS_WITH -> criteriaBuilder.like(criteriaBuilder.upper(stringExpression), escapedFilterValue.toUpperCase() + "%", EscapeCharacter.DEFAULT.getEscapeCharacter());
+                case EQUALS -> criteriaBuilder.equal(criteriaBuilder.upper(stringExpression), stringValue.toUpperCase());
+                default ->
+                        throw new UnsupportedOperationException("This type of filter is not supported for text data type");
+            };
+        }
+
+        if (ResourceFilter.DataType.NUMBER == filter.dataType()) {
+            Double valueDouble = Double.valueOf((String) value);
+            return switch (filter.type()) {
+                case NOT_EQUAL -> criteriaBuilder.notEqual(expression, valueDouble);
+                case LESS_THAN_OR_EQUAL -> criteriaBuilder.lessThanOrEqualTo((Expression<Double>) expression, valueDouble);
+                case GREATER_THAN_OR_EQUAL ->
+                        criteriaBuilder.greaterThanOrEqualTo((Expression<Double>) expression, valueDouble);
+                default ->
+                        throw new UnsupportedOperationException("This type of filter is not supported for number data type");
+            };
+        }
+        throw new IllegalArgumentException("The filter type " + filter.type() + " is not supported with the data type " + filter.dataType());
+
+    }
+
+    static String columnToDotSeparatedField(ResourceFilter.Column column) {
+        return switch (column) {
+            case SUBJECT_ID -> "subjectId";
+            case LIMIT -> "limit";
+            case LIMIT_NAME -> "limitName";
+            case LIMIT_TYPE -> "limitType";
+            case ACTUEL_OVERLOAD -> "actualOverload";
+            case UP_COMING_OVERLOAD -> "upComingOverload";
+            case OVERLOAD -> "overload";
+            case VALUE -> "value";
+            case SIDE -> "side";
+            case CONNECTED_COMPONENT_NUM -> "connectedComponentNum";
+            case SYNCHRONOUS_COMPONENT_NUM -> "synchronousComponentNum";
+            case STATUS -> "status";
+            case ITERATION_COUNT -> "iterationCount";
+            case SLACK_BUS_ID -> "slackBusId";
+            case SLACK_BUS_ID_ACTIVE_POWER_MISMATCH -> "slackBusActivePowerMismatch";
+            case DISTRIBUTED_ACTIVE_POWER -> "distributedActivePower";
```

Comment:
why to not use column.name();  ?

Reply:
I had added it before I changed the schema. I will clean up

Reply:
Done